monitor/src/main/java/edu/asu/diging/monitor/core/db/impl/AppDbConnection.java

```diff
@@ -47,12 +48,20 @@ public IApp store(IApp app) throws UnstorableObjectException {
         if (app.getId() == null) {
             throw new UnstorableObjectException("App does not have an id.");
         }
-        if (app.getRecipients() != null && !app.getRecipients().isEmpty()) {
-            for (INotificationRecipient recipient : app.getRecipients()) {
-                recipient.getApps().add((App) app);
-                em.merge(recipient);
-            }
-        } else {
+        if ((app.getTags() != null && !app.getTags().isEmpty()) || (app.getRecipients() != null && !app.getRecipients().isEmpty())) {

```

Comment:
I added this check to make sure the app is getting persisted only when both tags and recipients are both null/empty since we'll have to null check both tags and recipients and merge these entities when they're non-empty and persist the 'app' object only when both are empty

Reply:
but there might be changes to the app that need persisting independent of the tags or recipients, aren't there?

Reply:
Since tags/recipients have a many to many relation with the App entity, the application would throw an error that says an entity of App with the same id already exists in the persistence context (since we're adding the app entity to tag.getApps()/recipient.getApps()). This is the reason I added this check to make sure we're persisting the changes to the app only once.